### ISSUE TITLE
Njalla API change

### ISSUE THREAD

**Issue by arulagrawal** [Sighery/terraform-provider-njalla#61]
It appears that njalla silently changed their "id" parameter to be a string rather than int, which breaks most things that interact with it. (more details here: https://github.com/go-acme/lego/issues/1685)

It appears that this provider still creates the records but the output of `terraform apply` is filled with errors:

```
╷
│ Error: json: cannot unmarshal string into Go struct field Record.id of type int
│ 
│   with njalla_record_mx.mx,
│   on dns.tf line 73, in resource "njalla_record_mx" "mx":
│   73: resource "njalla_record_mx" "mx" {
│ 
╵

```
etc

**Comment by Sighery** [comment]
Ugh, this is a pain in the ass. It requires changing [the Go library](https://github.com/Sighery/gonjalla) as well, since this provider is really just a wrapper over that. The one good thing is that it seems infra might not need migrating as I initially thought, since Terraform already uses string IDs for keeping track of resources, and we were already converting the int IDs to str IDs in the provider, so here's to hoping this doesn't actually mean a breaking change in infra projects using this provider.

**Comment by Sighery** [comment]
Hey, thanks for reporting. This should be fixed with [the `v0.10.0` release](https://github.com/Sighery/terraform-provider-njalla/releases/tag/v0.10.0). If you haven't applied any changes it should be a painless update. Simply update the appropriate value in your Terraform infra:

```terraform
terraform {
  required_version = "~> 1.2"

  required_providers {
    njalla = {
      source  = "Sighery/njalla"
      version = "~> 0.10.0"
    }
  }
}
```

Then run in the console:

```sh
terraform init -upgrade
```

To update the lock file, and that should do the trick. The IDs should still be the same, which means when you go `plan` it won't throw any errors, and detect no changes so nothing will be shown for changes.

---

If you've already tried to apply changes, and the records were created but not assimilated into the Terraform state, you should be able to, after updating to v0.10.0, import the records with `terraform import njalla_record_txt.example-import example.com:12345`, where the resource address and domain/ID will depend on your specific setup. There's a [guide in the documentation on how to import](https://registry.terraform.io/providers/Sighery/njalla/0.10.0/docs/guides/importing-record-resources) which might be helpful. This link is generated off [docs/guides/importing-record-resources.md](https://github.com/Sighery/terraform-provider-njalla/blob/8639ecf8630518d211675d4c7d029a4aaf0af237/docs/guides/importing-record-resources.md).

If you tried to apply changes, and the records weren't created, then simply updating and trying to apply again should do the trick.

**Comment by arulagrawal** [comment]
Looks like its working. Thanks so much!
